lightning/src/chain/keysinterface.rs

```diff
@@ -233,13 +233,16 @@ pub trait ChannelKeys : Send+Clone {
 	// TODO: Document the things someone using this interface should enforce before signing.
 	fn sign_counterparty_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &CommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()>;
 
-	/// Create a signature for a holder's commitment transaction. This will only ever be called with
-	/// the same commitment_tx (or a copy thereof), though there are currently no guarantees
-	/// that it will not be called multiple times.
+	/// Create a signatures for a holder's commitment transaction and its claiming HTLC transactions.
+	/// This will only ever be called with a non-revoked commitment_tx.  This may be the latest
```

Comment:
I think we can be better on this comment.

"A signing entity (watchtower/ChannelMonitor replica) will only _initiate_ signing of a commitment tx for the latest local state. Due to a distributed setup, it might have to _react_ to a commitment tx broadcast and sign HTLC for the previous-latest local state." We should distinguish signing initiation from signing reaction.

W.r.t to "secondary ChannelMonitor", it designates one replica of a distributed ChannelMonitor setup, we should bind to one nomenclature and always call them "ChannelMonitor replica", as you might not have a "primary" at all in that kind of setup. 

If your HTLC state is empty, your Vec<Signature> might be empty, and that's completely fine.

"This may be called multiple times", I think the design purpose of this interface liberality is to handle reorg or concurrent broadcast, do we have other cases ?

Reply:
reworded, see if you like it.

yes, I think the multiple-times applies to reorgs, but it seems to me that fee bumps also cause re-signing if I understand the `OnChainTxHandler` code

Reply:
See the comment "Timer set to $NEVER given we can't bump tx without anchor outputs". This code path should execute only once, pre-anchor we can't bump holder's {commitment, htlc} txn. But right won't stay true for long.